File: tests/go/sort_stable_search/sort_strings_case_sensitive.go
// vybe-test: go/sort_stable_search/sort_strings_case_sensitive
// origin: languages/go/tests/go/test_sort_stable_search.rs

package main
import "fmt"
import "sort"
func __check(got string, want string) {
	if got != want {
		fmt.Println("FAIL: want [" + want + "] got [" + got + "]")
		panic("assertion failed")
	}
}

func main() { s := []string{"Banana", "apple", "Cherry"}
sort.Strings(s)
__check(fmt.Sprint(s[0]), "Banana")
__check(fmt.Sprint(s[2]), "apple") }

File: tests/js/array_sort_advanced/sort_strings_case_sensitive.js
// vybe-test: js/array_sort_advanced/sort_strings_case_sensitive
// origin: languages/js/tests/js/test_array_sort_advanced.rs

function __line(...args) {
    // console.log joins its arguments with a single space. String() is the
    // coercion Vybe's logging host applies to each one.
    return args.map(String).join(" ");
}

function __check(got, want) {
    if (got !== want) {
        console.log("FAIL: want [" + want + "] got [" + got + "]");
        throw new Error("assertion failed");
    }
}

const words = ["banana", "Apple", "cherry"];
words.sort();
// uppercase comes before lowercase in Unicode
__check(__line(words[0]), "Apple");
